lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2007/179d_ashrae_90_1_2007/179d_ashrae_90_1_2007.Model.rb

```
@@ -709,6 +749,209 @@ def model_create_prm_any_baseline_building(user_model, building_type, climate_zo
+                  
+                  # Get current OA flow rate for this air loop
+                  current_air_loop_oa = 0.0
+                  if controller_oa.minimumOutdoorAirFlowRate.is_initialized
+                    current_air_loop_oa = controller_oa.minimumOutdoorAirFlowRate.get
+                  elsif sizing_system.designOutdoorAirFlowRate.is_initialized
+                    current_air_loop_oa = sizing_system.designOutdoorAirFlowRate.get
+                  end
+                  
+                  # Calculate new OA flow rate for this air loop
+                  new_air_loop_oa = current_air_loop_oa * adjustment_factor
+                  
+                  # Set the new values - ensure maximum is at least equal to minimum to avoid EnergyPlus constraint violations
+                  sizing_system.setDesignOutdoorAirFlowRate(new_air_loop_oa)
+                  controller_oa.setMinimumOutdoorAirFlowRate(new_air_loop_oa)
+                  # Set maximum to be at least equal to minimum to satisfy EnergyPlus constraints
+                  if controller_oa.maximumOutdoorAirFlowRate.is_initialized
+                    current_max = controller_oa.maximumOutdoorAirFlowRate.get
+                    if current_max < new_air_loop_oa
+                      controller_oa.setMaximumOutdoorAirFlowRate(new_air_loop_oa)
+                    end
+                  else
+                    controller_oa.setMaximumOutdoorAirFlowRate(new_air_loop_oa)
+                  end
+                  
+                  OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.Model', 
+                    "Proportionally adjusted air loop '#{air_loop.nameString}': OA rate #{current_air_loop_oa.round(3)} -> #{new_air_loop_oa.round(3)} m³/s")
+                end
+              
+                OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.Model', "Applied proportional OA adjustment factor: #{adjustment_factor.round(4)}")
+              else
```

did you ever encounter our model getting into this `else` situation? it is nice to have something for the `else`, but technically, we should be putting the min OA rates based on certain requirement criteria which can sometimes be more than just floor-area-based criteria. also, maybe an easier option is to let it fail?!